### ISSUE TITLE
Diagnostics are cleared on file save

### ISSUE THREAD

**Issue by stormshield-guillaumed** [rust-lang/rust-analyzer#20440]
When I save a file in my editor (helix), the diagnostics are cleared, and no diagnostics are shown until I restart the LSP. After restart, they disappear again on file save.

I don't know if the problem only happens with helix or other editors aswell, but I don't have this problem with previous rust-analyzer versions. Maybe it's related to #20436.

Steps to reproduce:
1. `cargo new diag && cd diag`
2. `sed -i 's/println!("Hello, world!")/let n = 42/' src/main.rs`
3. `hx src/main.rs` 
4. Warning for unused variable is shown
5. Save with `:w`
6. Warning for unused variable is gone

**rust-analyzer version**: `rust-analyzer 0.3.2571-standalone`

**rustc version**: `rustc 1.91.0-nightly (1ebbd87a6 2025-08-11)`

**editor or extension**: `helix 25.07.1 (a05c151b)`

**relevant settings**: `check.command = "clippy"`

**code snippet to reproduce**:
```rust
fn main() {
    let n = 42;
}
```

**Comment by ShoyuVanilla** [comment]
This has been fixed in the latest stable release. Please try `0.3.2577`

**Comment by stormshield-guillaumed** [comment]
Thank you, sorry I didn't saw the hotfix as I only looked at the tag of the release which was the same date.
